tests/functionaltests/Tests/CoreAnimationTests/CALayerAppearanceTests.mm

```diff
@@ -0,0 +1,172 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+#import <TestFramework.h>
+#import <Foundation/Foundation.h>
+
+#include <COMIncludes.h>
+#import <WRLHelpers.h>
+#import <ErrorHandling.h>
+#import <RawBuffer.h>
+#import <wrl/client.h>
+#import <wrl/implements.h>
+#import <wrl/async.h>
+#import <wrl/wrappers/corewrappers.h>
+#import <windows.foundation.h>
+#include <COMIncludes_end.h>
+
+#import "ObjCXamlControls.h"
+
+#import "UWP/WindowsUIXamlControls.h"
+#import "CALayerInternal.h"
+
+// TODO: Consolidate this into a common place so that all tests can use it
+static const NSTimeInterval c_testTimeoutInSec = 5;
+
+@interface CALayerViewController : UIViewController
+@property CALayer* layer;
+@property UIView* viewForLayer;
+@end
+
+@implementation CALayerViewController
+- (instancetype)init {
+    if (self = [super init]) {
+        self->_layer = [CALayer layer];
+    }
+
+    return self;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    self->_viewForLayer = self.view;
+    self->_viewForLayer.frame = CGRectMake(0, 100, 200, 200);
+    self->_viewForLayer.backgroundColor = [UIColor whiteColor];
+
+    self->_layer.frame = self->_viewForLayer.bounds;
+    [self->_viewForLayer.layer addSublayer:self->_layer];
+}
+@end
+
+//
+// Validate that the CALayer opacity property change invokes a property change in the backing XAML element
+//
+TEST(CALayerAppearance, OpacityChanged) {
+    __block CALayerViewController* caLayerVC;
+    __block NSCondition* condition = [[[NSCondition alloc] init] autorelease];
+
+    dispatch_sync(dispatch_get_main_queue(), ^{
+        LOG_INFO("Creating CALayerViewController on the UI thread explicitly");
+        caLayerVC = [[CALayerViewController alloc] init];
+        [caLayerVC view];
+
+        // We have to artificially ref the element since the block needs to keep it around
+        WXUIElement* backingElement = [caLayerVC.layer _xamlElement];
+        [backingElement retain];
```

Comment:
Nice! Unfortunately, the scopexit doesn't like __block variables but we can still explicitly release it for now.

---
In reply to: [90351820](https://github.com/Microsoft/WinObjC/pull/1483#discussion_r90351820) [](ancestors = 90351820,90351052)